test/process/extract/ExtractorTest.cpp

```diff
@@ -0,0 +1,350 @@
+        // in the past. To test this, we start with a "lower" index entry and add a skip factor for the LAST test!
+
+        // Please also note, that the last line of the test dataset has NO newline, which can also occur in some source
+        // files. This line will not be found by the tested method here but will be output with a WARNING at the end of
+        // the extraction process.
+
+        for (int i = 0; i < 6; i++) {
+            auto startingLine = startingLines[i];
+            auto lineCount = 4;
+            auto indexEntry = indexEntries[indexEntryID[i]];
+            Extractor extractor(make_shared<PathSource>(fastq),
+                                make_shared<PathSource>(index),
+                                ConsoleSink::create(),
+                                false,
+                                ExtractMode::lines, startingLine, lineCount, 4, true);
+            extractor.calculateStartingLineAndLineCount(); // Needs to be done, otherwise startingLine and linecount are not set.
+            extractor.setSkip(startingLine - indexEntry->startingLineInEntry);
+
+            for (uint64_t j = startingBlockIDs[i]; j < _blockData.size(); j++) {
+                extractor.processDecompressedChunkOfData(_blockData[j], indexEntry->toIndexEntry());
+            }
+                    CHECK(extractor.getStoredLines().size() == expectedLines[i]);
+        }
+    }
+
+    /**
+     * This test will check the case where an incomplete line was printed because the last incomplete line was ignored.
+     * Conditions:
+     * - Still first pass, no lines written yet
+     * - Found index entry starting line offset == 0
+     * - Skip is 0 after the last invocation of processDecompressedData
+     * - lastIncomplete line is filled
+     */
+//    TEST ("Test special case: skip in first pass is of splitLines.size() with incomplete last line.") {
+//
+//    }
+
+    TEST (TEST_CREATE_EXTRACTOR_AND_EXTRACT_SMALL_TO_COUT) {
+        // We won't test the indexing here as it is already tested in TEST_CREATE_INDEX_SMALL
+        // What we do is:
+        // - Create the index
+        // - We decompress the fastq file with gunzip (which should work!)
+        // - We extract several line ranges and compare them with the help of head/tail with the original data
+        TestResourcesAndFunctions res(INDEXER_SUITE_TESTS, TEST_CREATE_EXTRACTOR_AND_EXTRACT_SMALL_TO_COUT);
+
+        path fastq = res.getResource(TEST_FASTQ_SMALL);
+        path index = res.filePath(TEST_INDEX_SMALL);
+        path extractedFastq = res.filePath("test.fastq");
+        u_int64_t linesInFastq = 4000;
+        vector<string> decompressedSourceContent;
+
+        if (!initializeComplexTest(fastq, index, extractedFastq, -1, linesInFastq, &decompressedSourceContent))
+            return;
+
+        // Do work, though the tests for the larger files don't.
```

Comment:
Why?
Or: Why should that be acceptible?

Should there not be a range test spanning the whole file?
What about the behaviour for an empty range?